dom/src/fromEvent.ts

```diff
@@ -30,3 +33,38 @@ export function fromEvent(
     } as Producer<Event>,
   );
 }
+
+export function preventDefaultConditional(
+  event: any,
+  preventDefault: PreventDefaultOpt,
+): void {
+  if (preventDefault) {
+    if (typeof preventDefault === 'boolean') {
+      event.preventDefault();
+    } else if (typeof preventDefault === 'function') {
+      if (preventDefault(event)) {
+        event.preventDefault();
+      }
+    } else if (typeof preventDefault === 'object') {
+      const matchObject: (m: {}, o: {}) => boolean = (matcher, obj) =>
```

Comment:
it's used twice, most noticably recursive here: https://github.com/cyclejs/cyclejs/pull/615/files/49431069da3dab1759b22c0253cf35949d2c69dc#diff-4974c1e7e3b97190ed437163ebe8d3adR54